src/main/scala/is/hail/expr/FunctionRegistry.scala

```diff
@@ -775,6 +810,55 @@ object FunctionRegistry {
     Construct a :ref:`genotype` object from a :ref:`call`.
     """)(callHr, genotypeHr)
 
+  registerSpecial("Genotype", { (v: () => Any, dosF: () => Any) =>
+    val dos = dosF()
+    Genotype(v().asInstanceOf[Variant].nAlleles, if (dos == null) null else dos.asInstanceOf[IndexedSeq[Double]].toArray) },
```

Comment:
Since this is a register special, the `v` can also be `null`. I'm not sure what should happen in that case, but I suspect a `NullPointerException` is not the right thing. Perhaps we should add a 
```
val vv = v()
if (vv == null)
  throw new HailException("the first argument to Genotype, the Variant, must not be NA")
```